esp/src/eclwatch/DFUQueryWidget.js

```diff
@@ -345,6 +345,30 @@ define([
 
             //  Implementation  ---
             getFilter: function () {
+                var context = this;
+                if (context.workunitsGrid){
+                    var retVal = this.filter.toObject();
+                        if (retVal.Sortby){
+                            switch (retVal.Sortby){
+                                case "Smallest":
+                                    context.workunitsGrid.set ("sort", [{ attribute: "FileSize", "descending": false }]);
+                                    break;
+                                case "Largest":
+                                    context.workunitsGrid.set ("sort", [{ attribute: "FileSize", "descending": true }]);
+                                    break;
+                                case "Oldest":
+                                    context.workunitsGrid.set ("sort", [{ attribute: "Modified", "descending": false }]);
+                                    break;
+                                case "Newest":
+                                    context.workunitsGrid.set ("sort", [{ attribute: "Modified", "descending": true }]);
+                                    break;
+                                default:
+                                    context.workunitsGrid.set ("sort", [{ attribute: "Modified", "descending": true }]);
```

Comment:
Very Minor:  The following might be cleaner:
```javascript
case "Newest":
default:
    context.workunitsGrid.set ("sort", [{ attribute: "Modified", "descending": true }]);
```

Reply:
Build failure on change, i think that is why i had separated the case and default previously, i will change it back to see if that is in fact the issue.

Reply:
I think that is just a lint issue.  Try:
```
case "Newest":
    /* falls through */
default:
    context.workunitsGrid.set ("sort", [{ attribute: "Modified", "descending": true }]);
```